File: src/main/java/org/example/Payment/BasePriceCalculator.java
package org.example.Payment;

public class BasePriceCalculator implements PriceCalculator {
    private int basePrice;

    public BasePriceCalculator(int basePrice) {
        this.basePrice = basePrice;
    }
    public int calculatePrice() {
        return this.basePrice;
    }
}

File: lib/services/base_price_calculator.dart
import '../models/rental.dart';
import 'price_calculator.dart';

class BasePriceCalculator implements PriceCalculator {
  final double basePrice;
  final int hours;

  BasePriceCalculator({
    required this.basePrice,
    required this.hours,
  });

  @override
  int calculate(Rental rental) {
    return (basePrice * hours).toInt();
  }
}
